connector/apmconnector/metric_connector.go

```diff
@@ -0,0 +1,89 @@
+// Copyright New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package apmconnector // import "apmconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type ApmMetricConnector struct {
+	config *Config
+	logger *zap.Logger
+
+	metricsConsumer consumer.Metrics
+}
+
+func (c *ApmMetricConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *ApmMetricConnector) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	metrics := ConvertTraces(c.logger, c.config, td)
+	err := c.metricsConsumer.ConsumeMetrics(ctx, metrics)
```

Comment:
[nitpick] as in logs this could return `c.metricsConsumer.ConsumeMetrics(ctx, metrics)`